### PR TITLE
SystemView: Fix Bus Hardfault when no JVM is running 

### PR DESCRIPTION
**Pull request by emoreac-microej**
Documentation update to provide a fixed implementation of `SYSVIEW_MICROEJ_X_OS_SendTaskList()`

YT: https://youtrack.cross/issue/M0119TD-27/SystemView-Bus-Hardfault-when-no-JVM-is-running

Fix tested in STM32F7508-DK: https://youtrack.cross/issue/P0234STM32F7508DK-261/Fix-hardfault-when-the-System-View-analysis-is-started-when-the-JVM-is-not-running

### CODE REVIEW

File: VEEPortingGuide/systemView.rst [hunk 1]
```diff
@@ -56,7 +56,9 @@ This file can be modified to fit your system configuration:
    * Update ``SYSVIEW_APP_NAME``, ``SYSVIEW_DEVICE_NAME``, and ``SYSVIEW_RAM_BASE`` defines to fit your system information.
    * To add MicroEJ Java threads management in SystemView tasks initialization:
   
-      * Add these includes ``#include "LLMJVM_MONITOR_SYSVIEW.h"`` and ``#include "LLTRACE_SYSVIEW_configuration.h"``.
+      * Add these includes ``#include "task.h"``, ``#include "LLMJVM_MONITOR_SYSVIEW.h"``, ``#include "LLTRACE_SYSVIEW_configuration.h"``, ``#include "SEGGER_SYSVIEW_configuration.h"``
+        and the include that declares the external variable ``pvMicrojvmCreatedTask``. ``pvMicrojvmCreatedTask`` must be the FreeRTOS task handle
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:60]:
```suggestion
        and the include that declares the external variable ``pvMEJCoreEngineTask``. ``pvMEJCoreEngineTask`` must be the FreeRTOS task handle
```

File: VEEPortingGuide/systemView.rst [hunk 2]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+
+         // The strategy to send tasks info is different in post mortem and live analysis.
+         #if (1 == SEGGER_SYSVIEW_POST_MORTEM_MODE)
+            /**
+            * POST MORTEM analysis
+            *
+            * In post mortem analysis, FreeRTOS tasks call regularly the function SYSVIEW_MICROEJ_X_OS_SendTaskList(),
+            * when a packet (systemview event) is sent to the SEGGER circular buffer. It is necessary because the information of tasks
+            * must be regularly uploaded in the circular buffer in order to provide a valid analysis at any moment.
+            * Consequently, we only allow to call LLMJVM_MONITOR_SYSTEMVIEW_send_task_list() when the current task is the Microjvm.
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:92]:
```suggestion
            * Consequently, we only allow to call LLMJVM_MONITOR_SYSTEMVIEW_send_task_list() when the current task is the MicroEJ Core Engine.
```

File: VEEPortingGuide/systemView.rst [hunk 3]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            */
+
+            /* Obtain the handle of the current task. */
+            TaskHandle_t xHandle = xTaskGetCurrentTaskHandle();
+            configASSERT( xHandle ); // Check the handle is not NULL.
+
+            // Check if the current task handle is the Microjvm task handle. pvMicrojvmCreatedTask is an external variable.
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:99]:
```suggestion
            // Check if the current task handle is the MicroEJ Core Engine task handle. pvMEJCoreEngineTask is an external variable.
```

File: VEEPortingGuide/systemView.rst [hunk 4]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            if( xHandle == pvMicrojvmCreatedTask){
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:100]:
```suggestion
            if( xHandle == pvMEJCoreEngineTask){
```

File: VEEPortingGuide/systemView.rst [hunk 5]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+               // Launched by the JVM, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
+               LLMJVM_MONITOR_SYSTEMVIEW_send_task_list();
+            }
+         #else
+            /**
+            * LIVE analysis
+            *
+            * In live analysis, the call of SYSVIEW_MICROEJ_X_OS_SendTaskList() is triggered by
+            * the SystemView Software through the J-Link probe. Consequently, the Microjvm task will never call
+            * the function LLMJVM_MONITOR_SYSTEMVIEW_send_task_list(). However, if the Microjvm task is created,
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:110]:
```suggestion
            * the function LLMJVM_MONITOR_SYSTEMVIEW_send_task_list(). However, if the MicroEJ Core Engine task is created,
```

File: VEEPortingGuide/systemView.rst [hunk 6]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            * the function must be called LLMJVM_MONITOR_SYSTEMVIEW_send_task_list().
+            */
+            // Check if the Microjvm task handle is not NULL. pvMicrojvmCreatedTask is an external variable.
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:113]:
```suggestion
            // Check if the MicroEJ Core Engine task handle is not NULL. pvMEJCoreEngineTask is an external variable.
```

File: VEEPortingGuide/systemView.rst [hunk 7]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            if( NULL != pvMicrojvmCreatedTask){
+               // The JVM task is running, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:115]:
```suggestion
               // The MicroEJ Core Engine task is running, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
```

File: VEEPortingGuide/systemView.rst [hunk 8]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            /**
+            * LIVE analysis
+            *
+            * In live analysis, the call of SYSVIEW_MICROEJ_X_OS_SendTaskList() is triggered by
+            * the SystemView Software through the J-Link probe. Consequently, the Microjvm task will never call
+            * the function LLMJVM_MONITOR_SYSTEMVIEW_send_task_list(). However, if the Microjvm task is created,
+            * the function must be called LLMJVM_MONITOR_SYSTEMVIEW_send_task_list().
+            */
+            // Check if the Microjvm task handle is not NULL. pvMicrojvmCreatedTask is an external variable.
+            if( NULL != pvMicrojvmCreatedTask){
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:114]:
```suggestion
            if( NULL != pvMEJCoreEngineTask){
```

File: VEEPortingGuide/systemView.rst [hunk 9]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            if( xHandle == pvMicrojvmCreatedTask){
+               // Launched by the JVM, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
+               LLMJVM_MONITOR_SYSTEMVIEW_send_task_list();
+            }
+         #else
+            /**
+            * LIVE analysis
+            *
+            * In live analysis, the call of SYSVIEW_MICROEJ_X_OS_SendTaskList() is triggered by
+            * the SystemView Software through the J-Link probe. Consequently, the Microjvm task will never call
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:109]:
```suggestion
            * the SystemView Software through the J-Link probe. Consequently, the MicroEJ Core Engine task will never call
```

File: VEEPortingGuide/systemView.rst [hunk 10]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
...
+            * Consequently, we only allow to call LLMJVM_MONITOR_SYSTEMVIEW_send_task_list() when the current task is the Microjvm.
+            */
+
+            /* Obtain the handle of the current task. */
+            TaskHandle_t xHandle = xTaskGetCurrentTaskHandle();
+            configASSERT( xHandle ); // Check the handle is not NULL.
+
+            // Check if the current task handle is the Microjvm task handle. pvMicrojvmCreatedTask is an external variable.
+            if( xHandle == pvMicrojvmCreatedTask){
+               // Launched by the JVM, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:101]:
```suggestion
               // Launched by the MicroEJ Core Engine, we execute LLMJVM_MONITOR_SYSTEMVIEW_send_task_list()
```

File: VEEPortingGuide/systemView.rst [hunk 11]
```diff
@@ -76,13 +78,44 @@ This file can be modified to fit your system configuration:
             LLMJVM_MONITOR_SYSTEMVIEW_send_task_list();
          }
          
-         void SEGGER_SYSVIEW_Conf(void) {
```
Comment (jcourcelmicroej) [inline VEEPortingGuide/systemView.rst:79]: Why this function has been removed ?

**Review by jcourcelmicroej — CHANGES_REQUESTED** [review]

File: VEEPortingGuide/systemView.rst (hunk 11 above)
Comment (emoreac-microej) [reply VEEPortingGuide/systemView.rst:79]: It's a mistake during the update, I will put it back.
Thanks for your attention.

**Comment by emoreac-microej** [comment]
Hi @jcourcelmicroej , thanks for the review.
I applied your feedback.

**Review by jcourcelmicroej — APPROVED** [review]

**Pull request merged**